client/tx/legacy.go

```diff
@@ -0,0 +1,64 @@
+package tx
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+)
+
+// ConvertTxToStdTx converts a transaction to the legacy StdTx format
+func ConvertTxToStdTx(codec *codec.Codec, tx sdk.Tx) (types.StdTx, error) {
```

Comment:
only the happy path is covered by tests

Reply:
Most of those failure paths are actually really hard to trigger. I'm just defensively returning errors that might occur. But actually triggering those cases feels way out of the scope of these tests - like passing in some Tx which doesn't implement `FeeGasTx` - it just simply doesn't exist in the SDK.

Reply:
But like I said I can try to cover at least amino and proto tx paths in the same build.